src/model/mod.rs

```diff
@@ -183,6 +186,21 @@ impl Serialize for CapabilityConfig {
     }
 }
 
+impl Hash for CapabilityConfig {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            CapabilityConfig::Json(v) => {
+                //NOTE(@ahmedtadde): By default, Serde uses BtreeMap for JSON map objects which is good for determinism.
+                //However, for arrays, ordering is not guaranteed.
+                //In general, unless the json values are strictly equal (ordering, casing, white spaces, etc...), the hash will be different. This is something to be mindful of when using this hash function.
+                let json = serde_json::to_string(v).expect("Should be able to serialize");
+                json.hash(state);
```

Comment:
in terms of fallback behavior, if the serialization fails, should the hash prefixing just be skipped OR should an error be returned as opposed to a full blown `panic`?

slightly related...is there an enforced size limit on the `provider_config` object? If not, should there be? i guess the same questions applies to the linkdef `values`. `"we'll deal with it if it becomes a problem"` is also a reasonable approach for now, I suppose.

Reply:
I don't know if you can return an error to hash, so it may just be good fallback behavior to default to some other value. 

For your related question, there isn't an enforced size limit so that may be an issue. We may want to use a hash method that uses a short hash value since we don't need to go from hash to values, we just want a uniqueness summary 🤔